src/chest.py

```diff
@@ -0,0 +1,88 @@
+import time
+import os
+
+from logger import Logger
+from template_finder import TemplateFinder
+from screen import Screen
+from char import IChar
+from config import Config
+
+class Chest:
+    def __init__(self, char: IChar, template_finder: TemplateFinder, template: str = None):
+        config = Config()
+        self._screen = Screen(config.general["monitor"])
+        self._char = char
+        self._template_finder = template_finder
+        self._folder_name = "chests"
+        self._min_score = 0.8
+        # load all templates
+        self._templates = []
+        for filename in os.listdir(f'assets/{self._folder_name}/{template}'):
+            filename = filename.lower()
+            if filename.endswith('.png'):
+                chest = filename[:-4].upper()
+                self._templates.append(chest)
+                
+    def open_up_chest(
+        self,  
+        time_out: float = 8,
+        threshold: float = 0.8
+    ) -> float:
+        score = 100
+        templates = self._templates
+        while len(templates) > 0:
+            template_match = self._template_finder.search(templates, self._screen.grab(), threshold=threshold)
+            score = template_match.score
+            if template_match.name is None:
+                break
+            if score <= threshold:
+                templates.remove(template_match.name)
+                continue
+            if template_match.valid:
+                Logger.debug(f"Opening {template_match.name} ({template_match.score*100:.1f}% confidence)")
+                x_m, y_m = self._screen.convert_screen_to_monitor(template_match.position)
+                # act as picking up a potion to support telekinesis
+                self._char.pick_up_item([x_m, y_m], 'potion')
```

Comment:
We might want to rename this, but for now the workaround is fine. Maybe just add a TODO comment to the pick_up_item function in i_char. e.g. `# TODO: Rename function as it is used for general object pick up / selection`.

Reply:
I think we might replace this with select_by_template(telekinesis=true)